src/main/java/org/jenkinsci/plugins/apperian/ApperianUpload.java

```diff
@@ -365,18 +374,12 @@ public ListBoxModel doFillApiTokenIdItems(@AncestorInPath Item job,
                     .includeCurrentValue(apiTokenId);
         }
 
+        @RequirePOST
         public FormValidation doCheckApiTokenId(@AncestorInPath Item job,
                                                 @QueryParameter("apiTokenId") final String apiTokenId) {
             // Verify that the user has permissions to access the Credentials
-            if (job == null) {
-                if (!Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER)) {
-                    return FormValidation.ok();
-                }
-            }
-            else {
-                if (!job.hasPermission(Item.EXTENDED_READ) && !job.hasPermission(CredentialsProvider.USE_ITEM)) {
-                    return FormValidation.ok();
-                }
+            if (!hasEnoughPermissions(job)) {
+                return FormValidation.error("missing permissions");
```

Comment:
So, to be sure I understand it. this behaviour has changed? Before it returns `FormValidation.ok();` and now `FormValidation.error("missing permissions");`. hasn't it?

Reply:
I don't think it was ok to say that validation is ok if you don't have enough permissions to validate, for that reason I have changed it. It is just a red message shown in the UI